src/libraries/Microsoft.Bcl.Memory/src/Microsoft.Bcl.Memory.csproj

```
@@ -27,6 +27,18 @@
     <Compile Include="$(CoreLibSharedDir)System\Buffers\Text\Base64Url\Base64UrlValidator.cs" Link="System\Buffers\Text\Base64Url\Base64UrlValidator.cs" />
   </ItemGroup>
 
+  <ItemGroup Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net8.0'))">
```

nit: condition on netframework/netstandard is more readable:
```suggestion
  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' or '$(TargetFrameworkIdentifier)' == '.NETStandard'">
```

---

It's just my personal preference, your condition works too but is a bit redundant since it will always be false for net8, net9, net10 since this package only targets net8+ (NetMinimum == net8.0 currently). This is different from Base64Url which we do want to include in the net8 build.

> I don't understand what needs to be fixed.

You can accept the suggestion I posted by clicking on "Commit suggestion", up to you.

---

```suggestion
  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
```

I'd suggest this, if we want to condition on `TargetFrameworkIdentifier`.

---

> It's just my personal preference, your condition works too but is a bit redundant since it will always be false for net8, net9, net10 since this package only targets net8+ (NetMinimum == net8.0 currently). This is different from Base64Url which we do want to include in the net8 build.

What if in the future the package is compiled for an older version of .NET, for example for .NET 5+? Then my condition will not need to be changed! It will continue to work correctly, so I would leave it redundant.

---

> What if in the future the package is compiled for an older version of .NET, for example for .NET 5+? 

We are not optimizing for this situation that is very unlikely to happen. We are actively deleting code for no longer supported versions from the repo, https://github.com/dotnet/runtime/pull/109176 is a recent example.